kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,1013 @@
+            if not forced_alpha and rgb[3] != 1.0:
+                attrib['line-opacity'] = str(rgb[3])
+            if linewidth != 1.0:
+                attrib['line-width'] = str(linewidth)
+            if self.get_joinstyle() != 'round':
+                attrib['line-linejoin'] = self.get_joinstyle()
+            if self.get_capstyle() != 'butt':
+                attrib['line-linecap'] = _capstyle_d[self.get_capstyle()]
+
+        return attrib
+
+
+def draw_if_interactive():
+    '''Handle whether or not the backend is in interactive mode or not.
+    '''
+    if matplotlib.is_interactive():
+        figManager = Gcf.get_active()
+        if figManager:
+            figManager.canvas.draw_idle()
+
+
+class Show(ShowBase):
+    '''mainloop needs to be overwritten to define the show() behavior for kivy
+       framework.
+    '''
+    def mainloop(self):
+        app = App.get_running_app()
+        if app:
+            app.run()
+
+show = Show()
+
+
+def new_figure_manager(num, *args, **kwargs):
+    '''Create a new figure manager instance for the figure given.
+    '''
+    # if a main-level app must be created, this (and
+    # new_figure_manager_given_figure) is the usual place to
+    # do it -- see backend_wx, backend_wxagg and backend_tkagg for
+    # examples. Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    '''Create a new figure manager instance for the given figure.
+    '''
+    canvas = FigureCanvasKivy(figure)
+    canvas.draw()
+    manager = FigureManagerKivy(canvas, num)
+    _create_App(canvas, manager.toolbar.actionbar)
+    return manager
+
+
+class FigureCanvasKivy(FocusBehavior, Widget, FigureCanvasBase):
+    '''FigureCanvasKivy class. See module documentation for more information.
+
+    .. versionadded:: 1.9.1
+    '''
+
+    def __init__(self, figure, **kwargs):
+        Window.bind(mouse_pos=self._on_mouse_pos)
+        self.bind(size=self._on_size_changed)
+        self.inside_figure = True
+        self.figure = figure
+        super(FigureCanvasKivy, self).__init__(figure=self.figure, **kwargs)
+        self.focus = True
+
+    def draw(self):
+        '''Draw the figure using the KivyRenderer
+        '''
+        self.clear_widgets()
+        self.canvas.clear()
+        renderer = RendererKivy(self)
+        self.figure.draw(renderer)
+
+    def on_touch_down(self, touch):
+        '''Kivy Event to trigger the following matplotlib events:
+           `motion_notify_event`, `scroll_event`, `button_press_event`,
+           `enter_notify_event` and `leave_notify_event`
+        '''
+        if super(FigureCanvasKivy, self).on_touch_down(touch):
+            return True
+        FigureCanvasBase.motion_notify_event(self, touch.x,
```

Comment:
Do you have actual tests showing these coordinates are correct. I'm suspicious that the coordinates you're passing will not be what mpl events are expecting. They are in kivy coordinates. Also, the inside figure logic seems wrong and doesn't seem to work in test_backend for example.